tribe-ext-export-mec-events.php

```diff
@@ -0,0 +1,362 @@
+			), 'imports', 'tribe_aggregator_disable', false );
+		}
+
+		/**
+		 * Make a button to trigger the CSV creation process
+		 *
+		 * @param string $text
+		 *
+		 * @return string
+		 */
+		public function export_button( $type, $text = '' ) {
+			$text     = $text ? $text : __( 'Export Events', 'the-events-calendar' );
+			$settings = Tribe__Settings::instance();
+
+			// get the base settings page url
+			$url = apply_filters( 'tribe_settings_url', add_query_arg( array(
+				'post_type' => Tribe__Events__Main::POSTTYPE,
+				'page'      => $settings->adminSlug,
+				'tab'       => 'imports',
+			), admin_url( 'edit.php' ) ) );
+
+			$url = add_query_arg( array( $type => '1' ), $url );
+			$url = wp_nonce_url( $url, $type );
+
+			return sprintf( '<a href="%s" class="button">%s</a>', $url, $text );
+		}
+
+		/**
+		 * If the button is clicked, start working
+		 *
+		 */
+		public function listen_for_export_button() {
+			if ( empty( $_REQUEST['export_events'] ) && empty( $_REQUEST['export_organizers'] ) && empty( $_REQUEST['export_venues'] ) || ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_events' ) && ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_organizers' ) && ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_venues' ) ) {
+				return;
+			} elseif ( ! empty( $_REQUEST['export_events'] ) || wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_events' ) ) {
+				$this->events_csv_setup();
+			} elseif ( ! empty( $_REQUEST['export_organizers'] ) || wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_organizers' ) ) {
+				$this->organizers_csv_setup();
+			} elseif ( ! empty( $_REQUEST['export_venues'] ) || wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_venues' ) ) {
+				$this->venues_csv_setup();
+			}
+		}
+
+		/**
+		 * Get the event data from Modern Events Calendar
+		 *
+		 * @return array $event_data
+		 */
+		public function get_event_data() {
+
+			$event_data = array();
+
+			global $wpdb;
+
+			$events = $wpdb->get_results( "SELECT t1.post_title,
+       				t1.post_content,
+       				t2.start,
+       				t2.end,
+       				t2.time_start,
+       				t2.time_end,
+       				t3.meta_value AS organizer_id,
+       				t4.meta_value AS venue_id,
+
+  					(SELECT sum(meta_value)
+   						FROM wp_postmeta AS t5
+   						WHERE t1.ID = t5.post_id
+     						AND t5.meta_key = 'mec_cost') AS cost
+				  FROM `wp_posts` AS t1
+				  INNER JOIN wp_mec_events AS t2
+				  INNER JOIN wp_postmeta AS t3
+				  INNER JOIN wp_postmeta AS t4
+				  WHERE t1.ID = t2.post_id
+  					AND t1.ID = t3.post_id
+  					AND t1.ID = t4.post_id
+  					AND t3.meta_key = 'mec_organizer_id'
+  					AND t4.meta_key = 'mec_location_id'" );
```

Comment:
My only concern here is that without a limit clause, we could end up fetching more events than we have memory to process (if the calendar had a lot of MEC events). We can always iterate if there turns out to be a need.